### ISSUE TITLE
Add Image metadata in `WriteImage`

### ISSUE THREAD

**Issue by fdawg4l** [vmware/vic#195]
We need to persist a bit of metadata (in this case, some json which composes the last bit of history of that image retrieved from the hub) with each image.  The use case `docker history <id>`, which returns each layer of the `<id>` as composed by the `Dockerfile` history.

For instance
```
fahmed@hasselhoff: ~/src/vic/src/github.com/vmware/vic$ docker images | grep 55bb0ee25994
bootstrap-d3050f4       latest              55bb0ee25994        2 weeks ago         1.002 GB

fahmed@hasselhoff: ~/src/vic/src/github.com/vmware/vic$ docker history 55bb0ee25994
IMAGE               CREATED             CREATED BY                                      SIZE                COMMENT
55bb0ee25994        2 weeks ago         /bin/sh -c #(nop) CMD ["cat" "/tmp/bootstrap.   0 B                 
b4db80955044        2 weeks ago         |4 BUILD_ID=2016/02/12_@_21:09:38@master:d305   70.65 MB            
a988c926f4f0        2 weeks ago         |4 BUILD_ID=2016/02/12_@_21:09:38@master:d305   24.02 MB            
2e07594dd570        2 weeks ago         /bin/sh -c #(nop) COPY dir:2685afd8e32c13e051   112 MB              
bc6051cb9deb        3 weeks ago         /bin/sh -c #(nop) ENV DEBUG=1                   0 B                 
e459851f8374        3 weeks ago         /bin/sh -c #(nop) ENV VIC=/go/src/github.com/   0 B                 
55ff2372727f        3 weeks ago         /bin/sh -c #(nop) ENV GOPATH=/go                0 B                 
da9f10e21762        3 weeks ago         /bin/sh -c #(nop) ARG TINYCORE_DEPS             0 B                 
96b1cc938ecb        3 weeks ago         /bin/sh -c #(nop) ARG TINYCORE                  

...<snip>...

251f1f3f88aa        7 weeks ago         /bin/sh -c apt-get update && apt-get install    134 MB              
215e4fe64afc        7 weeks ago         /bin/sh -c apt-get update && apt-get install    122.6 MB            
706c9f1cbd8c        7 weeks ago         /bin/sh -c apt-get update && apt-get install    44.29 MB            
d4b2ba78e3b4        7 weeks ago         /bin/sh -c #(nop) CMD ["/bin/bash"]             0 B                 
cb6fb082434e        7 weeks ago         /bin/sh -c #(nop) ADD file:0098703cdfd5b5eda3   125.1 MB
```

If each of these layers (`Image`s) is the result of it's `parent` + some command from the Dockerfile, we should be able to traverse the lineage of an image on the datastore to its parent and rebuild its history.

The solution is to add this bit of metadata (the cmd that composes that image) with each image (as a file in the image's directory in the datastore or maybe as something which can be queried via vmomi).

cc @caglar10ur @hickeng

**Comment by fdawg4l** [comment]
This is pretty much done.  Will post a PR shortly.

**Comment by fdawg4l** [comment]
Fixed in a3ec0374c823576335cb7a5cff1acdb40dfb3aa9
